pq-src/build.rs

```diff
@@ -181,90 +185,96 @@ int main() {"#,
 make_test_for!(TEST_FOR_STRCHRNUL, r#"strchrnul("", 42);"#);
 make_test_for!(TEST_FOR_STRSIGNAL, r#"strsignal(32);"#);
 
-fn check_compiles(test: &str) -> bool {
-    let test_path = std::env::var("OUT_DIR").expect("Set by cargo") + "/test.c";
-    std::fs::write(&test_path, &test).expect("Failed to write test");
-    let r = cc::Build::new().file(&test_path).try_compile("test");
-    std::fs::remove_file(test_path).expect("Failed to remove test file");
+fn check_compiles(test: &str, mut command: cc::Build) -> bool {
+    // Add necessary compiler-flags to make sure that undefined functions actually cause errors
+    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
+    let target_env = env::var("CARGO_CFG_TARGET_ENV").unwrap();
+    match (target_env.as_str(), target_os.as_str()) {
```

Comment:
There is also [`Build::is_flag_supported`](https://docs.rs/cc/latest/cc/struct.Build.html#method.is_flag_supported) which might allow to just simplify that logic by trying if that flag is supported and then adding it regardless of the environment?

Reply:
Even better (imo) there is [Build::flag_if_supported](https://docs.rs/cc/latest/cc/struct.Build.html#method.flag_if_supported) which I have used in e4557ce